### ISSUE TITLE
Tensorflow training refactor

### ISSUE THREAD

**Issue by Error323** [glinscott/leela-chess#41]
Refactor the training to support multiple GPUs (each training a different model for now) and automatic decaying learning rate. All main() entry points should get a yaml config file as argument that carries relevant information e.g.

```yaml
name: 'My Network'
gpu: 0
dataset: '/path/to/dataset'
training:
    batch_size: 1024
    learning_rate: 0.1
    decay_rate: 0.1
    decay_step: 100000
    policy_loss_weight: 1.0
    value_loss_weight: 0.01
model:
   filters: 64
   residual_blocks: 5
```
It would then be useful to add the yaml file to the `weights.txt.gz` at [this repository](https://github.com/glinscott/lczero-weights).
